src/Material.cpp

```diff
@@ -663,38 +435,400 @@ void Material::setNumEnergyGroups(const int num_groups) {
-               "%d which contains %d energy groups", num_groups, _id, _num_groups);
+  /* If the data has already been aligned, do nothing */
+  if (_data_aligned)
+    return;
 
-  for (int i=0; i < _num_groups; i++)
-    _sigma_t[i] = FP_PRECISION(xs[i]);
-}
+  if (_num_groups <= 0)
+    log_printf(ERROR, "Unable to align Material %d data since the "
+               "cross-sections have not yet been set\n", _id);
+
+  _num_vector_groups = (_num_groups / VEC_LENGTH) + 1;
+
+  /* Allocate memory for the new aligned xs data */
+  int size = _num_vector_groups * VEC_LENGTH * sizeof(FP_PRECISION);
+
+  /* Allocate word-aligned memory for cross-section data arrays */
+  FP_PRECISION* new_sigma_t = (FP_PRECISION*)MM_MALLOC(size, VEC_ALIGNMENT);
+  FP_PRECISION* new_sigma_a = (FP_PRECISION*)MM_MALLOC(size, VEC_ALIGNMENT);
+  FP_PRECISION* new_sigma_f = (FP_PRECISION*)MM_MALLOC(size, VEC_ALIGNMENT);
+  FP_PRECISION* new_nu_sigma_f=(FP_PRECISION*)MM_MALLOC(size, VEC_ALIGNMENT);
+  FP_PRECISION* new_chi = (FP_PRECISION*)MM_MALLOC(size, VEC_ALIGNMENT);
+
+  /* Initialize data structures to ones for sigma_t since it is used to
+   * divide the source in the solver, and zeroes for everything else */
+  size = _num_vector_groups * VEC_LENGTH * sizeof(FP_PRECISION);
+  for (int i=0; i < _num_vector_groups * VEC_LENGTH; i++) {
+    new_sigma_t[i] = 1.0;
+    new_sigma_a[i] = 0.0;
+    new_sigma_f[i] = 0.0;
+    new_nu_sigma_f[i] = 0.0;
+    new_chi[i] = 0.0;
+  }
+
+  /* Copy materials data from unaligned arrays into new aligned arrays */
+  size = _num_groups * sizeof(FP_PRECISION);
+  memcpy(new_sigma_t, _sigma_t, size);
+  memcpy(new_sigma_a, _sigma_a, size);
+  memcpy(new_sigma_f, _sigma_f, size);
+  memcpy(new_nu_sigma_f, _nu_sigma_f, size);
+  memcpy(new_chi, _chi, size);
+
+  /* Delete the old unaligned arrays */
+  delete [] _sigma_t;
+  delete [] _sigma_a;
+  delete [] _sigma_f;
+  delete [] _nu_sigma_f;
+  delete [] _chi;
+
+  /* Set the material's array pointers to the new aligned arrays */
+  _sigma_t = new_sigma_t;
+  _sigma_a = new_sigma_a;
+  _sigma_f = new_sigma_f;
+  _nu_sigma_f = new_nu_sigma_f;
+  _chi = new_chi;
+
+  _data_aligned = true;
+
+  return;
+}
+
+/* ************************************************************************** */
+
+
+/**
+ * @brief Constructor sets the ID and unique ID for the MacroMaterial.
+ * @param id the user-specified optional Material ID
+ * @param name the user-specified optional Material name
+ */
+MacroMaterial::MacroMaterial(int id, const char* name) : Material(id,name) {
+    
+  _material_type = MACRO_MATERIAL;
+  
+  _sigma_s = NULL;
+  _dif_coef = NULL;
+  _dif_hat = NULL;
+  _dif_tilde = NULL;
+  _buckling = NULL;
+  
+  return;
+}
+
+
+/**
+ * @brief Destructor deletes all cross-section data structures from memory.
+ */
+MacroMaterial::~MacroMaterial() {
+  if (_data_aligned) {
+    if (_sigma_s != NULL)
+      MM_FREE(_sigma_s);
+  }
+  
+  else {
```

Comment:
You also need to deallocate memory for the ``_dif_coeff``, ``_dif_hat``, ``_dif_tilde``, and ``_buckling`` arrays.